### ISSUE TITLE
feat: Collection Comparison View

### ISSUE THREAD

**Issue by aGallea** [aGallea/embedding-clusters#15]
## Problem

Users can only visualize one collection at a time. If you indexed both `fashion_imageUrl` (image embeddings) and `fashion_productDisplayName` (text embeddings), there is no way to see how image-based clusters compare to text-based clusters for the same underlying data.

This is a missed insight opportunity: seeing where visual similarity and semantic similarity agree or diverge is genuinely valuable for understanding data.

## Proposed Solution

Add a comparison mode that overlays two collections in the same 3D space. Run dimensionality reduction on the combined embedding space (with collection-of-origin labels), then visualize with distinct visual encodings per collection. Highlight where clusters agree and where they diverge.

### Backend

**New endpoint in: `embedding_cluster/server/routes/plot.py`**
- `POST /api/plot/compare`
  - Request:
    ```json
    {
      "collection_a": "fashion_imageUrl",
      "collection_b": "fashion_productDisplayName",
      "num_clusters": 10,
      "reduction_algorithm": "tsne",
      "match_field": "id"
    }
    ```
  - Logic:
    1. Fetch embeddings from both collections
    2. Match items by `match_field` (items present in both collections)
    3. For visualization, two approaches:
       a. **Shared space**: Concatenate both embedding sets, run t-SNE/UMAP on the combined set with a `source` label. This shows how the same items land differently under each embedding.
       b. **Procrustes alignment**: Run t-SNE/UMAP on each separately, then align via Procrustes rotation.
    4. Run k-means on each collection's embeddings separately
    5. Return: combined point data with `{ id, x, y, z, collection_source, cluster_id_a, cluster_id_b }`
  - Response: `CompareResponse` with point arrays, cluster data, and agreement metrics

**Modify: `embedding_cluster/scatter_plot.py`**
- Add `compare_collections()` function that handles the combined reduction pipeline
- Compute agreement metrics:
  - **Adjusted Rand Index (ARI)**: How much the two clusterings agree
  - **Normalized Mutual Information (NMI)**: Information shared between clusterings
  - **Agreement rate**: % of item pairs that are in the same cluster in both collections

**Modify: `embedding_cluster/server/models.py`**
- Add `CompareRequest` and `CompareResponse` models
- Include agreement metrics in response

### Frontend

**New file: `frontend/src/components/plot/CompareControls.tsx`**
- Two collection dropdowns (A and B)
- Match field selector
- Shared cluster count
- "Compare" button

**New file: `frontend/src/components/plot/CompareView.tsx`**
- 3D scatter plot showing overlaid points
- Visual encoding for collection source:
  - Option 1: Different shapes (spheres vs cubes)
  - Option 2: Different opacity (solid vs semi-transparent)
  - Option 3: Color-coded border/outline
- Lines connecting the same item across collections (shows how it "moved" between embedding spaces)
- Toggle to show/hide connection lines
- Agreement metric cards (ARI, NMI, agreement %)

**Modify: `frontend/src/pages/PlotPage.tsx`**
- Add "Compare" tab alongside the regular plot view
- Switch between single-collection and comparison mode

**Modify: `frontend/src/stores/plotStore.ts`**
- Add compare state: `compareCollections`, `compareResults`, `showConnections`
- Add `compareCollections()` action

**Modify: `frontend/src/api/plot.ts`**
- Add `compareCollections()` API function

### Visualization Options

The comparison view is complex — start with the simplest approach:
1. **Phase 1**: Side-by-side 3D views (same camera, synchronized rotation)
2. **Phase 2**: Overlaid view with visual encoding
3. **Phase 3**: Connection lines between matched items

Consider implementing Phase 1 first and iterating.

## Testing Requirements

Full-coverage tests are required for all new code.

### Backend Tests

**New file: `tests/test_compare.py`**
- Test compare endpoint with two valid collections
- Test matching items by ID field
- Test with collections that have partial overlap (some items in A but not B)
- Test with collections that have no overlap (error)
- Test agreement metrics (ARI, NMI) with known synthetic data
- Test that dimensionality reduction runs on combined data
- Test invalid collection names return 404
- Test compare response structure

**Modify: `tests/test_scatter_plot.py`**
- Test `compare_collections()` function with synthetic embeddings
- Test agreement metrics computation
- Test Procrustes alignment (if implemented)

### Frontend Tests
- CompareControls: dual collection selectors, compare button
- CompareView: renders overlaid points, connection lines toggle, metrics display
- PlotPage: tab switching between single and compare modes
- Store: compare state management

## Acceptance Criteria

- [ ] User can select two collections to compare
- [ ] System matches items by a shared ID field
- [ ] Both collections are visualized in a combined 3D space
- [ ] Visual encoding distinguishes which collection each point belongs to
- [ ] Agreement metrics (ARI, NMI) show how much the clusterings agree
- [ ] Items present in both collections are visually connected
- [ ] All new backend code has full test coverage
- [ ] All new frontend components have tests

## Notes

- After implementation, update `README.md` to document the comparison feature
- This is the most complex visualization feature — consider phased implementation
- ARI and NMI are available from `sklearn.metrics` (existing dependency)
- Start with shared-space t-SNE approach (simplest) before adding Procrustes alignment
